build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/ProcessLogcatTiming.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public class ProcessLogcatTiming : Task
+	{
+		public string Logcat { get; set; }
+
+		public string Application { get; set; }
+
+		public string ResultsFile { get; set; }
+
+		public override bool Execute ()
+		{
+			using (var reader = new StreamReader (Logcat)) {
+				string line;
+				int pid = -1;
+				var procStartRegex = new Regex ($"^(\\d+-\\d+\\s+[\\d\\:\\.]+)\\s+.*ActivityManager: Start proc.*for added application {Application}\\: pid=(\\d+)");
+				Regex timingRegex = null;
+				var runtimeInitRegex = new Regex ("Runtime\\.init\\: end native-to-managed");
+				DateTime start = DateTime.Now;
+				DateTime last = start;
+				DateTime initEnd = start;
+
+				while ((line = reader.ReadLine ()) != null) {
+					if (pid == -1) {
+						var match = procStartRegex.Match (line);
+						if (match.Success) {
```

Comment:
Please take a page from [mono's coding guidelines](http://www.mono-project.com/community/contributing/coding-guidelines), in particular the [indentation section](http://www.mono-project.com/community/contributing/coding-guidelines/#indentation), and *invert* this check:

```csharp
if (!match.Success)
    continue;
start = ParseTime (match.Groups [1].Value);
...
```

This helps keep the indentation from getting unwieldy.